File: src/main/java/com/roomj/simpletitles/TitleSettingsState.java
package com.roomj.simpletitles;

import com.intellij.openapi.application.ApplicationManager;
import com.intellij.openapi.components.PersistentStateComponent;
import com.intellij.openapi.components.State;
import com.intellij.openapi.components.Storage;
import com.intellij.util.xmlb.XmlSerializerUtil;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

@State(name = "com.roomj.simpletitles.TitleSettingsState", storages = @Storage("SdkSettingsPlugin.xml"))
public class TitleSettingsState implements PersistentStateComponent<TitleSettingsState> {

    public String projectFormat = "{DEFAULT}";
    public String fileFormat = "{SIMPLE}";

    public static TitleSettingsState getInstance() {
        return ApplicationManager.getApplication().getService(TitleSettingsState.class);
    }

    @Nullable
    @Override
    public TitleSettingsState getState() {
        return this;
    }

    @Override
    public void loadState(@NotNull TitleSettingsState state) {
        XmlSerializerUtil.copyBean(state, this);
    }
}

File: src/main/kotlin/com/roomj/simpletitles/TitleSettingsState.kt
package com.roomj.simpletitles

import com.intellij.openapi.application.ApplicationManager
import com.intellij.openapi.components.PersistentStateComponent
import com.intellij.openapi.components.State
import com.intellij.openapi.components.Storage
import com.intellij.util.xmlb.XmlSerializerUtil

@State(name = "com.roomj.simpletitles.TitleSettingsState", storages = [Storage("SdkSettingsPlugin.xml")])
class TitleSettingsState : PersistentStateComponent<TitleSettingsState> {

    var projectFormat: String = "{DEFAULT}"
    var fileFormat: String = "{SIMPLE}"

    override fun getState(): TitleSettingsState = this

    override fun loadState(state: TitleSettingsState) {
        XmlSerializerUtil.copyBean(state, this)
    }

    companion object {
        fun getInstance(): TitleSettingsState =
            ApplicationManager.getApplication().getService(TitleSettingsState::class.java)
    }
}